packages/ag-charts-enterprise/src/error-bar/errorBar.ts

```diff
@@ -0,0 +1,138 @@
+
+const { ChartAxisDirection, Validate, OPT_STRING } = _ModuleSupport;
+
+type CartesianSeries<
+    C extends _ModuleSupport.SeriesNodeDataContext<any, any>,
+    N extends _Scene.Node
+> = _ModuleSupport.CartesianSeries<C, N>;
+
+export interface ErrorBarPoints {
+    readonly yLowerPoint: _Scene.Point;
+    readonly yUpperPoint: _Scene.Point;
+}
+
+export class ErrorBarNode extends _Scene.Path {
+    public points: ErrorBarPoints = { yLowerPoint: { x: 0, y: 0 }, yUpperPoint: { x: 0, y: 0 } };
+
+    updatePath() {
+        // TODO(olegat) implement, this is just placeholder draw code
+        const { path } = this;
+        this.fill = 'black';
+        this.stroke = 'black';
+        this.strokeWidth = 1;
+        this.fillOpacity = 1;
+        this.strokeOpacity = 1;
+
+        const { yLowerPoint, yUpperPoint } = this.points;
+
+        path.clear();
+        path.moveTo(yLowerPoint.x, yLowerPoint.y);
+        path.lineTo(yUpperPoint.x, yUpperPoint.y);
+        path.closePath();
+    }
+}
+
+export class ErrorBars extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
+    @Validate(OPT_STRING)
+    yLowerKey: string = '';
+
+    @Validate(OPT_STRING)
+    yLowerName?: string = undefined;
+
+    @Validate(OPT_STRING)
+    yUpperKey: string = '';
+
+    @Validate(OPT_STRING)
+    yUpperName?: string = undefined;
+
+    private readonly cartesianSeries: _ModuleSupport.CartesianSeries<any, any>;
+    private readonly groupNode: _Scene.Group;
+    private readonly selection: _Scene.Selection<ErrorBarNode>;
+    private nodeData: (ErrorBarPoints | undefined)[] = [];
+
+    constructor(ctx: _ModuleSupport.SeriesContext) {
+        super();
+
+        const supportedSeriesTypes = ['line'];
+        if (!supportedSeriesTypes.includes(ctx.series.type)) {
+            throw new Error(
+                `AG Charts - unsupported series type '${
+                    ctx.series.type
+                }', error bars supported series types: ${supportedSeriesTypes.join(', ')}`
+            );
+        }
+        this.cartesianSeries = ctx.series as CartesianSeries<any, any>;
+        const { contentGroup } = this.cartesianSeries;
+
+        this.groupNode = new _Scene.Group({ name: `${contentGroup.id}-series-errorBars` });
+        contentGroup.appendChild(this.groupNode);
+        this.selection = _Scene.Selection.select(this.groupNode, () => this.errorBarFactory());
+
+        const listener = this.cartesianSeries.addListener('data-model', (event) => this.onDataProcessed(event));
+        this.destroyFns.push(() => this.cartesianSeries.removeListener(listener));
+    }
+
+    onDataProcessed(event: {
+        dataModel: _ModuleSupport.DataModel<any, any, any>;
+        processedData: _ModuleSupport.ProcessedData<any>;
+    }) {
+        if (event.dataModel !== undefined && event.processedData !== undefined) {
+            this.createNodeData(event.dataModel, event.processedData);
+            this.update();
```

Comment:
I don't think it's a major issue, but this does mean that the `ErrorBars` `createNodeData()` and `update()` happen during the series `processData` phase, which seems a little inconsistent.

Reply:
`createNodeData` is perhaps a misnomer. `CartesianSeries.createNodeData` is called by [`CartesianSeries.updateSeries`](https://github.com/ag-grid/ag-charts/blob/4286604b832df7197d966ec609c89a81fe3e0a42/packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts#L296) which is called by [`CartesianSeries.update`](https://github.com/ag-grid/ag-charts/blob/4286604b832df7197d966ec609c89a81fe3e0a42/packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts#L272). So this module consistent with the series that uses it.